internal/pkg/admission/policy-server-ca-secret.go

```diff
@@ -115,29 +128,34 @@ func (r *Reconciler) fetchOrInitializePolicyServerCARootSecret(ctx context.Conte
 		)
 		return nil, errors.Join(errors.New("cannot fetch or initialize Policy Server CA secret"), err)
 	}
+
 	setTrueConditionType(
 		&policyServer.Status.Conditions,
 		string(policiesv1.PolicyServerCARootSecretReconciled),
 	)
+
 	return policyServerSecret, nil
 }
 
-func updateSecretCA(policyServerSecret *corev1.Secret, generateCA generateCAFunc, pemEncodeCertificate pemEncodeCertificateFunc) error {
-	caRoot, err := generateCA()
+func updateCASecret(policyServerSecret *corev1.Secret) error {
```

Comment:
This function is always generating the CA, it doesn't look into the contents of the secret that is passed as parameter.

Why don't we change the signature of the function to be something like that?

```golang
func createInternalCASecret() (policyServerSecret *corev1.Secret, error)
```